### PR TITLE
feat: WAYF displays a name of the service

### PR DESCRIPTION
**Pull request by BaranekD**

### CODE REVIEW

File: lib/Disco.php [hunk 1]
```diff
@@ -115,6 +118,9 @@ public function __construct(
             $id = explode(":", $query[self::AUTH_ID])[0];
             $state = State::loadState($id, self::SAML_SP_SSO, true);
 
+            $translate = new Translate(Configuration::getInstance());
+            $this->spName = $translate->getPreferredTranslation($state[self::STATE_SP_METADATA][self::NAME]);
```
Comment (vyskocilpavel) [inline lib/Disco.php:122]: $state[self::STATE_SP_METADATA][self::NAME] can be null.

File: lib/Disco.php [hunk 2]
```diff
@@ -844,4 +874,43 @@ private static function constructSearchData($idpMetadata): string
...
+                return $item;
+            }
+        }
+
+        return null;
+    }
+
+    private function fillSpName($state)
+    {
+        $translate = new Translate(Configuration::getInstance());
```
Comment (dBucik) [inline lib/Disco.php:891]: Please make use of the translation available in the template. Pass the template into the function and use it like `$t->t('{key}')`

File: lib/Disco.php [hunk 3]
```diff
@@ -844,4 +874,43 @@ private static function constructSearchData($idpMetadata): string
...
+
+        $clientIdWithPrefix = self::substrInArray(self::CLIENT_ID_PREFIX, $this->originalAuthnContextClassRef);
+
+        if ($clientIdWithPrefix !== null) {
+            $parts = explode(':', $clientIdWithPrefix);
+            $clientId = end($parts);
+
+            $this->adapter = Adapter::getInstance($this->wayfConfiguration->getString(self::INTERFACE, self::RPC));
+
+            $facility = $this->adapter->getFacilityByClientId($clientId);
```
Comment (dBucik) [inline lib/Disco.php:901]: Just to be sure, I would probably catch any exceptions thrown by the method. If we do not have the name, disco should work anyway.

**Review by dBucik — CHANGES_REQUESTED** [review]

**Comment by dBucik** [comment]
@vyskocilpavel can you take a look and leave some feedback?

**Pull request merged**